metadata/etcd_test.go

```diff
@@ -689,46 +690,76 @@ func TestEtcdConfig_Put(t *testing.T) {
 	}
 }
 
-func TestEtcdConfig_Get(t *testing.T) {
+func TestEtcdConfig_InvalidServer(t *testing.T) {
 	if testing.Short() {
 		t.Skip()
 	}
-	type fields struct {
-		Host string
-		Port string
+	config := EtcdConfig{[]EtcdNode{{Host: "localhost", Port: ""}}}
+	_, err := config.InitClient()
+	if err == nil {
+		t.Errorf("InitClient() should have failed with invalid server")
 	}
-	type args struct {
-		key string
+}
+
+func TestEtcdConfig_GetEmptyKey(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
 	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		want    []byte
-		wantErr bool
-	}{
-		{"Test Invalid Server", fields{"localhost", ""}, args{key: ""}, nil, true},
-		{"Test Invalid Key", fields{"localhost", "2379"}, args{key: "testkey"}, []byte{}, false},
+	config := EtcdConfig{[]EtcdNode{{Host: "localhost", Port: "2379"}}}
+	c, err := config.InitClient()
+	if err != nil {
+		t.Errorf("InitClient() could not initialize client: %v", err)
 	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			config := EtcdConfig{[]EtcdNode{{Host: tt.fields.Host, Port: tt.fields.Port}}}
-			c, err := config.InitClient()
-			if err != nil && !tt.wantErr {
-				t.Errorf("Get() could not initialize client: %v", err)
-			} else if err != nil && tt.wantErr {
-				return
-			}
-			client := EtcdStorage{c}
-			got, err := client.Get(tt.args.key)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("Get() error = %v, wantErr %v", err, tt.wantErr)
-				return
-			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Fatalf("Get() got = %v, want %v", got, tt.want)
-			}
-		})
+	client := EtcdStorage{c}
+	_, err = client.Get("")
+	if err == nil {
+		t.Errorf("Get() should have failed with invalid key")
+	}
+	// check to see if it's an etcderror
+	if _, ok := err.(rpctypes.EtcdError); !ok {
+		t.Errorf("Get() should have failed with etcd error")
+	}
+}
+
+func TestEtcdConfig_GetNonExistentKey(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+	config := EtcdConfig{[]EtcdNode{{Host: "localhost", Port: "2379"}}}
+	c, err := config.InitClient()
+	if err != nil {
+		t.Errorf("InitClient() could not initialize client: %v", err)
+	}
+	client := EtcdStorage{c}
+	_, err = client.Get("non_existent")
+	if err == nil {
+		t.Errorf("Get() should have failed with invalid key")
+	}
+	// check to see if it's an etcderror
+	if _, ok := err.(*KeyNotFoundError); !ok {
+		t.Errorf("Get() should have failed with KeyNotFoundError")
+	}
+}
+
+func TestEtcdConfig_ValidKey(t *testing.T) {
```

Comment:
LGTM but we can change the tests into test cases instead of separate tests. For now, we are good.

Reply:
why split it up?

Reply:
why do test cases?

Reply:
I actually tried to it at first. They're slightly different enough where it actually makes the one bigger test harder to read.

Reply:
let me try one more time

Reply:
alright i think i made it better